query-engine/prisma-models/src/builders/model_builder.rs

```diff
@@ -40,11 +40,12 @@ impl ModelBuilder {
         let pk = self.primary_key.map(|pk| pk.build(&all_fields));
         let fields = Fields::new(all_fields, Arc::downgrade(&model), pk);
 
-        let indexes = self.indexes.into_iter().map(|i| i.build(&fields.scalar())).collect();
+        let indexes = self.indexes.into_iter().map(|i| i.build(&fields.all)).collect();
```

Comment:
I'm kind of torn. It seems ok here. But it could also be moved into a `Impl From` for indexes